### PR TITLE
Added support for AzureGermanCloud

### PR DESCRIPTION
**Pull request by horrion**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [ ] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by hovsepm** [comment]
@azuresdkci test this please

**Comment by hovsepm** [comment]
@horrion Hi Robert. Thank you for the contribution! We would like to better understand your usage scenarios because  Authentication library in this repo is not supported anymore. We moved all the  functionality to azure-powershell repo because it was used primarily there.

**Comment by markcowl** [comment]
Note that, if you wanted to make a contribution that impacts the current libraries, an update here:  https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/src/ClientRuntime/Microsoft.Rest.ClientRuntime.Azure.Authentication/ActiveDirectoryServiceSettings.cs would be accepted

**Comment by shahabhijeet** [comment]
@horrion add to whitelist

**Comment by shahabhijeet** [comment]
@azuresdkci add to whitelist

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

**Comment by horrion** [comment]
@markcowl please be advised that I just updated the content of https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/src/ClientRuntime/Microsoft.Rest.ClientRuntime.Azure.Authentication/ActiveDirectoryServiceSettings.cs

File: src/Authentication/Common.Authentication/Models/AzureEnvironment.Methods.cs [hunk 1]
```diff
@@ -143,6 +143,32 @@ private string StorageFileEndpointFormat()
                         {AzureEnvironment.Endpoint.GraphEndpointResourceId, AzureEnvironmentConstants.USGovernmentGraphEndpoint}
                     }
                 }
+            },
```
Comment (markcowl) [inline src/Authentication/Common.Authentication/Models/AzureEnvironment.Methods.cs]: @horrion We have no plans to update this library, so please revert this.

File: src/ClientRuntime/Microsoft.Rest.ClientRuntime.Azure.Authentication/ActiveDirectoryServiceSettings.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public sealed class ActiveDirectoryServiceSettings
 
         private static readonly ActiveDirectoryServiceSettings AzureSettings = new ActiveDirectoryServiceSettings
         {
-            AuthenticationEndpoint= new Uri("https://login.windows.net/"), 
+            AuthenticationEndpoint= new Uri("https://login.microsoftonline.com/"),
```
Comment (markcowl) [inline src/ClientRuntime/Microsoft.Rest.ClientRuntime.Azure.Authentication/ActiveDirectoryServiceSettings.cs]: login.windows.net is preferred.

**Review by markcowl — CHANGES_REQUESTED** [review]

**Review by markcowl — CHANGES_REQUESTED** [review]
Thanks for this PR, I love to see this kinf of proactive support.  We are deprecating the AutoRest branch, but once this is ready, either you or we can move this into the new branch.

Also, need to update the library version so that we cna publish a new nuget package with the changes.

Thanks again for the contribution!

**Comment by markcowl** [comment]
@horrion Note that the ci failure is due to adding the new environment in common.authentication without updating one of the tests - since this part of the change should be reverted, the tests should pass once this is done.

File: src/ClientRuntime/Microsoft.Rest.ClientRuntime.Azure.Authentication/ActiveDirectoryServiceSettings.cs (hunk 2 above)
Comment (horrion) [inline src/ClientRuntime/Microsoft.Rest.ClientRuntime.Azure.Authentication/ActiveDirectoryServiceSettings.cs]: @markcowl Could you please explain why the "old" endpoint is preferred? I thought everything that depends on Azure Resource Manager has to go through the new endpoint (login.microsoftonline.com).

**Comment by horrion** [comment]
@markcowl I'll take care of removing the new environment in common.authentication. 
Would you mind notifying me when the new branch becomes available?

File: src/ClientRuntime/Microsoft.Rest.ClientRuntime.Azure.Authentication/ActiveDirectoryServiceSettings.cs (hunk 2 above)
Comment (markcowl) [inline src/ClientRuntime/Microsoft.Rest.ClientRuntime.Azure.Authentication/ActiveDirectoryServiceSettings.cs]: @horrion You are absolutely right, I was reading this entirely the wrong way

**Comment by horrion** [comment]
@markcowl Thanks for your support, so far! I have just created a new PR for the vs17Dev branch here: https://github.com/Azure/azure-sdk-for-net/pull/3182

**Pull request closed without merging**